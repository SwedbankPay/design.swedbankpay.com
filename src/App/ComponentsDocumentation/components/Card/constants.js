import React from "react";
import CardsComponent from "@components/Cards";

const BASENAME = process.env.basename;

export const overviewCards = {
    id: "overviewCards",
    tabsId: "overviewButtonsTabs",
    elements: [
        {
            tab: "Standard",
            component: <CardsComponent
                titleTxt="Title"
                type="primary"
                text="Write the supportive paragraph here."
                icon={<i className="material-icons">shopping_basket</i>}
            />,
            options: {
                radio: [
                    {
                        id: "addons_radio",
                        title: "Addons",
                        values: [
                            {
                                name: "Plain",
                                value: {
                                    icon: null
                                }
                            },
                            {
                                name: "With icon",
                                default: true,
                                value: {
                                    icon: <i className="material-icons">shopping_basket</i>
                                }
                            },
                            {
                                name: "With image",
                                value: {
                                    imgSrc: `${BASENAME}img/documentation/cards/img-example.png`,
                                    icon: null
                                }
                            }
                        ]
                    },
                    {
                        id: "color_radio",
                        title: "Accent color",
                        values: [
                            {
                                name: "Yellow brown",
                                value: {
                                    type: "primary"
                                }
                            },
                            {
                                name: "Turquoise",
                                value: {
                                    type: "secondary"
                                }
                            },
                            {
                                name: "Purple",
                                value: {
                                    type: "tertiary"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Standard cards",
            description: "The standard card component can be customized either icon or picture in order to make it easier for users to scan the content. Mixing simple card without icon should mostly be done when seperating the cards functionality such as when there are more cards to be displayed or when approperiate icons can not be found."
        },
        {
            tab: "Wide",
            component: <CardsComponent
                titleTxt="Title"
                type="primary"
                text="Write the supportive paragraph here."
                wide
                icon={<>01</>}
            />,
            options: {
                radio: [
                    {
                        id: "addons_radio",
                        title: "Addons",
                        values: [
                            {
                                name: "Plain",
                                value: {
                                    icon: null
                                }
                            },
                            {
                                name: "With icon",
                                default: true,
                                value: {
                                    icon: <>01</>
                                }
                            }
                        ]
                    },
                    {
                        id: "color_radio",
                        title: "Accent color",
                        values: [
                            {
                                name: "Yellow brown",
                                value: {
                                    type: "primary"
                                }
                            },
                            {
                                name: "Turquoise",
                                value: {
                                    type: "secondary"
                                }
                            },
                            {
                                name: "Purple",
                                value: {
                                    type: "tertiary"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Wide cards",
            description: "The wide card component can be used with and without numbering. Try to avoid mixing different card setups with each other as it could be confusing for the users to understand. Try to keep the title concise and to the point, to make it simple to understand and the design aligned with the cards as a group."
        }
    ]
};
