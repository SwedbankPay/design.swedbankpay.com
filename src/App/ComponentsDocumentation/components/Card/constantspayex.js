import React from "react";
import CardsComponent from "@components/Cards";

const defaultOptionsValues = {
	isButton: false,
	isWide: false,
	iconClasses: "at-document",
	hasTitle: true,
	imgSrc: null,
	imgRatio: null,
	hasTextContent: true,
	hasCTAText: true,
};

// Happy Woman image

const brand = process.env.brand;

const imgSrcURL =
	brand === "payex"
		? "https://design.payex.com/v/10.10.1/img/payex/documentation/cards/happy-woman.png"
		: "https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg";

export const overviewCards = {
	id: "overviewCards",
	tabsId: "overviewButtonsTabs",
	elements: [
		{
			tab: "Standard",
			component: (
				<CardsComponent
					titleTxt="Title"
					textContent="Write the supportive paragraph here."
					hasCTAText={defaultOptionsValues.hasCTAText}
					isButton={defaultOptionsValues.isButton}
					isWide={false}
					hasTitle={defaultOptionsValues.hasTitle}
					imgSrc={defaultOptionsValues.imgSrc}
					iconClasses={defaultOptionsValues.iconClasses}
					hasTextContent={defaultOptionsValues.hasTextContent}
					imgRatio={defaultOptionsValues.imgRatio}
				/>
			),
			options: {
				radio: [
					{
						id: "title",
						title: "Title",
						values: [
							{
								name: "Text only",
								value: {
									iconClasses: null,
									hasTitle: true,
								},
							},
							{
								name: "Text + icon",
								default: true,
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: true,
								},
							},
							{
								name: "Icon only",
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: false,
								},
							},
						],
					},
					{
						id: "has_image",
						title: "Image illustration",
						values: [
							{
								name: "No image",
								default: true,
								value: {
									imgSrc: null,
								},
							},
							{
								name: "With image - 16/9 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-16-9",
								},
							},
							{
								name: "With image - 4/3 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-4-3",
								},
							},
							{
								name: "With image - 1/1 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-1-1",
								},
							},
						],
					},
					{
						id: "text_content",
						title: "Text content",
						values: [
							{
								name: "With text content",
								default: true,
								value: {
									hasTextContent: true,
								},
							},
							{
								name: "No text content",
								value: {
									hasTextContent: false,
								},
							},
						],
					},
					{
						id: "has_cta_text",
						title: "CTA text",
						values: [
							{
								name: "With CTA text",
								default: true,
								value: {
									hasCTAText: true,
								},
							},
							{
								name: "Only arrow",
								value: {
									hasCTAText: false,
								},
							},
						],
					},
					{
						id: "card_element_tag",
						title: "Card purpose",
						values: [
							{
								name: "action <button>",
								value: {
									isButton: true,
								},
							},
							{
								name: "navigation link <a>",
								default: true,
								value: {
									isButton: false,
								},
							},
						],
					},
				],
			},
			title: "Standard cards",
			description:
				"The standard card component can be customized either icon or picture in order to make it easier for users to scan the content. Mixing simple card without icon should mostly be done when seperating the cards functionality such as when there are more cards to be displayed or when approperiate icons can not be found.",
		},

		{
			tab: "Wide",
			component: (
				<CardsComponent
					titleTxt="Title"
					textContent="Write the supportive paragraph here."
					hasCTAText={defaultOptionsValues.hasCTAText}
					isButton={defaultOptionsValues.isButton}
					isWide={true}
					hasTitle={defaultOptionsValues.hasTitle}
					imgSrc={defaultOptionsValues.imgSrc}
					iconClasses={defaultOptionsValues.iconClasses}
					hasTextContent={defaultOptionsValues.hasTextContent}
					imgRatio={defaultOptionsValues.imgRatio}
				/>
			),
			options: {
				radio: [
					{
						id: "title",
						title: "Title",
						values: [
							{
								name: "Text only",
								value: {
									iconClasses: null,
									hasTitle: true,
								},
							},
							{
								name: "Text + icon",
								default: true,
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: true,
								},
							},
							{
								name: "Icon only",
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: false,
									hasCTAText: true,
								},
							},
						],
					},
					{
						id: "has_image",
						title: "Image illustration",
						values: [
							{
								name: "No image",
								default: true,
								value: {
									imgSrc: null,
								},
							},
							{
								name: "With image",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: null,
								},
							},
						],
					},
					{
						id: "text_content",
						title: "Text content",
						values: [
							{
								name: "With text content",
								default: true,
								value: {
									hasTextContent: true,
								},
							},
							{
								name: "No text content",
								value: {
									hasTextContent: false,
								},
							},
						],
					},
					{
						id: "has_cta_text",
						title: "CTA text",
						values: [
							{
								name: "With CTA text",
								default: true,
								value: {
									hasCTAText: true,
								},
							},
							{
								name: "Only arrow",
								value: {
									hasCTAText: false,
								},
							},
						],
					},
					{
						id: "card_element_tag",
						title: "Card purpose",
						values: [
							{
								name: "action <button>",
								value: {
									isButton: true,
								},
							},
							{
								name: "navigation link <a>",
								default: true,
								value: {
									isButton: false,
								},
							},
						],
					},
				],
			},
			title: "Wide cards",
			description:
				"The wide card component can be customized either icon or picture in order to make it easier for users to scan the content. Try to avoid mixing different card setups with each other as it could be confusing for the users to understand. Try to keep the title concise and to the point, to make it simple to understand and the design aligned with the cards as a group.",
		},
	],
};