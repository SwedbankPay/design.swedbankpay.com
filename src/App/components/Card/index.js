import React from "react";
import PropTypes from "prop-types";

const Card = ({ type, titleTxt, titleMarkup, imgSrc, text, textSection, smallText, btn, btnClass, btnText, bodyClass, footerTxt, footerLink, footerLinkTxt, children }) => (
    <div className={`card${type ? ` card-${type}` : ""}`}>
        {imgSrc ? <div className="card-img">{"\n"}
            <img src={imgSrc} alt=""/>{"\n"}
        </div> : null}
        {titleTxt
            ? (<header className="card-header">
                <h3>{titleTxt}</h3>
            </header>)
            : (titleMarkup
                ? (<header className="card-header">
                    {titleMarkup}
                </header>)
                : null)}
        <div className={`card-body${bodyClass ? ` ${bodyClass}` : ""}`}>
            {children ? children : null }
            {text ? <p>{text}</p> : null}{"\n"}
            {textSection ?
                textSection.map((secText, i) => <p key={`card-p-${i}`}>{secText}</p>)
                : null
            }
            {btn ? <button type="button" className={`btn btn-executive${btnClass ? ` ${btnClass}` : ""}`}>{"\n"}{btnText}{"\n"}</button> : null}
        </div>
        {footerTxt || footerLinkTxt
            ? <footer className="card-footer">
                {smallText ? <small>{smallText}</small> : null}
                {footerTxt ? footerTxt : null}
                {footerLinkTxt ? <div className="footer-link"><a href={footerLink || "#"}>{footerLinkTxt}</a></div> : null}
            </footer> : null}
    </div>
);

Card.propTypes = {
    type: PropTypes.oneOf(["primary", "secondary", "plain"]),
    imgSrc: PropTypes.string,
    titleTxt: PropTypes.string,
    titleMarkup: PropTypes.object,
    icon: PropTypes.string,
    text: PropTypes.string,
    smallText: PropTypes.string,
    textSection: PropTypes.array,
    btn: PropTypes.bool,
    btnText: PropTypes.string,
    footerLink: PropTypes.string,
    footerLinkTxt: PropTypes.string
};

export default Card;
