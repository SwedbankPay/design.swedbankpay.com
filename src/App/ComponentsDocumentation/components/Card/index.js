import React, { Component } from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, DocContainer, StatusBadge } from "@docutils";
import CardComponent from "@components/Card";
import MediaObject from "@components/MediaObject";
import CodeTags from "@components/CodeTags";
import { overviewCards } from "./constants";

const textArr = ["This is a lot of text", "With some more text", "And then even some more", "Is it really possible to have this much text in one card?", "Yes!"];
const BASENAME = process.env.basename;
const brand = process.env.brand;

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <ComponentPreview language="html" showCasePanel showCasePanelAdvanced={overviewCards} codeFigure />

        <h3>{"Dos and dont's"}</h3>
        <div className="row">
            <div className="col-xl-6 mb-4">
                <div className="slab slab-plain slab-border-success h-100">
                    <h4>Do</h4>
                    When presenting a group of cards, make sure their color match and have the same size and text length
                    <img src={`${BASENAME}img/documentation/cards/do-cards.png`} className="w-100 mt-4 mb-4"/>
                </div>
            </div>
            <div className="col-xl-6 mb-4">
                <div className="slab slab-plain slab-border-error h-100">
                    <h4>{"Don't"}</h4>
                    Avoid miss-match colors, different sizes and length on text in order to present a clean view
                    <img src={`${BASENAME}img/documentation/cards/dont-cards.png`} className="w-100 mt-4 mb-4"/>
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col-xl-6 mb-4">
                <div className="slab slab-plain slab-border-success h-100">
                    <h4>Do</h4>
                    When presenting a group of cards, make sure they match visually. Try keeping it the same text length and to one row when possible.
                    <img src={`${BASENAME}img/documentation/cards/do-wide-cards.png`} className="w-100 mt-4 mb-4"/>
                </div>
            </div>
            <div className="col-xl-6 mb-4">
                <div className="slab slab-plain slab-border-error h-100">
                    <h4>{"Don't"}</h4>
                    Avoid miss-match colors, different sizes and length on text in order to present a clean view.
                    When using numbers make sure the numbering is sequenced and follows the correct order.
                    <img src={`${BASENAME}img/documentation/cards/dont-wide-cards.png`} className="w-100 mt-4 mb-4"/>
                </div>
            </div>
        </div>
    </>
);

const OldCard = () => (
    <>
        <h2 id="old-card">Old card</h2>
        <StatusBadge type="remove" />
        <p>
            Cards support a wide variety of content, including images, text, list groups, links, and more.
            Make sure to restrain the width of your card unless you wish for it to fill the wrapping component.
        </p>
        <ComponentPreview language="html" showCasePanel showCasePanelSm codeFigure>
            <CardComponent
                titleTxt="Card title"
                type="primary"
                imgSrc="https://picsum.photos/300/?random"
                text="This paragraph contains some text related to this card."
                footerTxt="Footer text"
            />
        </ComponentPreview>
        <h3>Card with grid</h3>
        <p>
            Use our grid along with cards to control their size and how they are displayed on different screen resolutions.
            Make sure to add <CodeTags type="secondary" code=".d-flex" /> to the wrapper if you want the cards to be of equal height when next to each other.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="row">
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        text="Card text"
                        smallText="Small card text"
                        btn
                        btnText="Button"
                        btnClass="btn-block mt-auto"
                        bodyClass="d-flex flex-column"
                    >
                        <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                    </CardComponent>
                </div>

                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        textSection={textArr}
                        smallText="This is some small text"
                        btn
                        btnText="Button"
                        btnClass="btn-block mt-auto"
                        bodyClass="d-flex flex-column"
                    />
                </div>

                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        imgSrc="https://picsum.photos/300/?random"
                        text="This is a card with an image and body"
                        footerLinkTxt="Footer link"
                    />
                </div>
            </div>
        </ComponentPreview>
        <h3>Card Types</h3>
        <p>
            We offer three different card types; <CodeTags type="secondary" code=".card-primary" />, <CodeTags type="secondary" code=".card-secondary" /> and <CodeTags type="secondary" code=".card-plain" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="row">
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="Card Primary"
                        imgSrc="https://picsum.photos/300/?random"
                        text="This paragraph contains some text about the person displayed above."
                    >
                        <ul className="list">
                            <li><span className="font-weight-bold">Card number:</span> 4563 5648 5642</li>
                            <li><span className="font-weight-bold">Balance:</span> 2500,- </li>
                        </ul>
                    </CardComponent>
                </div>
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="secondary"
                        titleTxt="Card Secondary"
                        text="Descriptive text to put inside the card, could contain a lot."
                        footerLinkTxt="Footer link"
                    >
                        <MediaObject size="sm" icon="account_balance" heading="725 NOK" text="Outstanding balance" textSmall muted />
                    </CardComponent>
                </div>
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="plain"
                        titleTxt="Card Plain"
                        imgSrc="https://picsum.photos/300/?random"
                        text="Descriptive text to put inside the card, could contain a lot."
                        footerLinkTxt="Footer link"
                    >
                        <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                    </CardComponent>
                </div>
            </div>
        </ComponentPreview>
    </>
);

class Card extends Component {
    render () {
        return (
            <DocContainer docToc>
                <p>
                    The card component is a flexible container that groups and informs about the content and action.
                    The information presented should be concise, to the point and easy to understand. The card
                    component should be used to direct to pages on our site and not to external sites, in that case
                    use <Link to="/components/links">Links</Link> instead.
                </p>
                {brand === "swedbankpay" && <Overview />}
                <OldCard />
            </DocContainer>
        );
    }
}

export default Card;

/* For testing */
export { Overview, OldCard };
