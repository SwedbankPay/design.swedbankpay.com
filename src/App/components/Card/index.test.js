import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Card from "./index";

describe("Component: Card -", () => {
    it("is defined", () => {
        expect(Card).toBeDefined();
    });

    it("throws an error if type does not match proptypes", () => {
        console.error = jest.fn();

        render(<Card type="error" />);

        expect(console.error).toHaveBeenCalled();

        const componentForSnap = renderer.create(<Card type="error" />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders with only body when no props are passed", () => {
        const { container } = render(<Card />);

        expect(screen.queryByRole("banner")).toBeNull();
        expect(container.querySelector(".card-img")).toBeFalsy();
        expect(container.querySelector(".card-body")).toBeTruthy();
        expect(screen.queryByRole("footer")).toBeNull();

        const componentForSnap = renderer.create(<Card />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders with class name card", () => {
        const { container } = render(<Card />);

        expect(container.querySelector("div")).toHaveClass("card");

        const componentForSnap = renderer.create(<Card />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders with correct class name", () => {
        const { container } = render(<Card type="brand"/>);

        expect(container.querySelector("div")).toHaveClass("card-brand");
        // FIXME:  why would they have it? delete ?
        expect(container.querySelector("div")).not.toHaveClass("card-default");
        expect(container.querySelector("div")).not.toHaveClass("card-light");

        const componentForSnap = renderer.create(<Card type="brand"/>);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders a title", () => {
        render(<Card titleTxt="Card title goes here"/>);

        expect(screen.getByRole("banner")).toBeInTheDocument();
        expect(screen.getByRole("heading")).toHaveTextContent("Card title goes here");

        const componentForSnap = renderer.create(<Card titleTxt="Card title goes here"/>);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders an image", () => {
        const { container } = render(<Card imgSrc="url.address" />);

        expect(container.querySelector("card-img")).toBeNull();
        expect(screen.getByRole("img")).toHaveAttribute("src", "url.address");

        const componentForSnap = renderer.create(<Card imgSrc="url.address" />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders a text in card-body", () => {
        const { container } = render(<Card text="Text that goes in card body" />);

        expect(container.querySelectorAll("p")).toBeTruthy();
        expect(container.querySelectorAll("p")).toHaveLength(1);
        expect(container.querySelector("p")).toHaveTextContent("Text that goes in card body");

        const componentForSnap = renderer.create(<Card text="Text that goes in card body" />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders a child object", () => {
        const { container } = render(<Card>This is child text</Card>);

        expect(container.querySelector(".card-body")).toBeInTheDocument();
        expect(container.querySelector(".card-body")).toHaveTextContent("This is child text");

        const componentForSnap = renderer.create(<Card>This is child text</Card>);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders with a button", () => {
        render(<Card btn />);

        expect(screen.getByRole("button")).toBeTruthy();
        expect(screen.getAllByRole("button").length).toBe(1);

        const componentForSnap = renderer.create(<Card btn />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders footer text", () => {
        const { container } = render(<Card footerTxt="Footer information" />);

        expect(screen.getByRole("contentinfo")).toHaveClass("card-footer");
        expect(container.querySelectorAll(".card-footer").length).toBe(1);
        expect(screen.getByRole("contentinfo")).toHaveTextContent("Footer information");

        const componentForSnap = renderer.create(<Card footerTxt="Footer information" />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });

    it("renders footer link text", () => {
        const { container } = render(<Card footerLinkTxt="Click this footer link" />);

        expect(container.querySelectorAll(".footer-link").length).toBe(1);
        expect(container.querySelector(".footer-link a")).toBeInTheDocument();
        expect(screen.getByRole("link")).toHaveTextContent("Click this footer link");

        const componentForSnap = renderer.create(<Card footerLinkTxt="Click this footer link" />);

        expect(componentForSnap.toJSON()).toMatchSnapshot();
    });
});
