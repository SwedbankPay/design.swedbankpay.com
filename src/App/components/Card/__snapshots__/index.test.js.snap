// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card -  renders a child object 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    This is child text
    

  </div>
</div>
`;

exports[`Component: Card -  renders a text in card-body 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <p>
      Text that goes in card body
    </p>
    

  </div>
</div>
`;

exports[`Component: Card -  renders a title 1`] = `
<div
  className="card"
>
  <header
    className="card-header"
  >
    <h3>
      Card title goes here
    </h3>
  </header>
  <div
    className="card-body"
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders an image 1`] = `
<div
  className="card"
>
  <div
    className="card-img"
  >
    

    <img
      alt=""
      src="url.address"
    />
    

  </div>
  <div
    className="card-body"
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders footer link text 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    

  </div>
  <footer
    className="card-footer"
  >
    <div
      className="footer-link"
    >
      <a
        href="#"
      >
        Click this footer link
      </a>
    </div>
  </footer>
</div>
`;

exports[`Component: Card -  renders footer text 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    

  </div>
  <footer
    className="card-footer"
  >
    Footer information
  </footer>
</div>
`;

exports[`Component: Card -  renders with a button 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    

    <button
      className="btn btn-primary"
      type="button"
    >
      

      

    </button>
  </div>
</div>
`;

exports[`Component: Card -  renders with class name card 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders with correct class name 1`] = `
<div
  className="card card-brand"
>
  <div
    className="card-body"
  >
    

  </div>
</div>
`;

exports[`Component: Card -  renders with only body when no props are passed 1`] = `
"<div class=\\"card\\"><div class=\\"card-body\\">
</div></div>"
`;

exports[`Component: Card -  throws an error if type does not match proptypes 1`] = `
<div
  className="card card-error"
>
  <div
    className="card-body"
  >
    

  </div>
</div>
`;
